From 523621574aff31d7949b241da12884e91e9dac75 Mon Sep 17 00:00:00 2001
From: Vladimir 'phcoder' Serbinenko <phcoder@gmail.com>
Date: Wed, 24 Apr 2013 15:07:24 +0200
Subject: [PATCH 341/482] 	* grub-core/normal/menu_text.c (print_entry):
 Put an asterisk 	in front of chosen entry to mark it even if
 highlighting is lost.

---
 ChangeLog                    | 5 +++++
 grub-core/normal/menu_text.c | 4 +++-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index 5854326..7e29788 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,10 @@
 2013-04-24  Vladimir Serbinenko  <phcoder@gmail.com>
 
+	* grub-core/normal/menu_text.c (print_entry): Put an asterisk
+	in front of chosen entry to mark it even if highlighting is lost.
+
+2013-04-24  Vladimir Serbinenko  <phcoder@gmail.com>
+
 	* grub-core/loader/i386/linux.c (grub_linux_boot): Default to
 	gfxpayload=keep if cbfb is active.
 
diff --git a/grub-core/normal/menu_text.c b/grub-core/normal/menu_text.c
index 0031b0c..e1d3c8f 100644
--- a/grub-core/normal/menu_text.c
+++ b/grub-core/normal/menu_text.c
@@ -242,7 +242,7 @@ print_entry (int y, int highlight, grub_menu_entry_t entry,
 	|| unicode_title[i] == '\r' || unicode_title[i] == '\e')
       unicode_title[i] = ' ';
 
-  for (x = GRUB_TERM_LEFT_BORDER_X + GRUB_TERM_MARGIN + 1, i = 0;
+  for (x = GRUB_TERM_LEFT_BORDER_X + GRUB_TERM_MARGIN + 2, i = 0;
        x < (int) (GRUB_TERM_LEFT_BORDER_X + grub_term_border_width (term)
 		  - GRUB_TERM_MARGIN);)
     {
@@ -269,6 +269,8 @@ print_entry (int y, int highlight, grub_menu_entry_t entry,
 	break;
     }
 
+  grub_putcode (highlight ? '*' : ' ', term);
+
   grub_print_ucs4 (unicode_title,
 		   unicode_title + last_printed, 0, 0, term);
 
-- 
1.8.2.1

